src/converters/helpers.ts

```diff
@@ -22,14 +23,112 @@ export function nullResult(...messages: Message[]): MemberResult {
   };
 }
 
-export function isCatalogMember(m: any): m is CatalogMember {
-  return is.string(m?.type) && is.string(m?.name);
+export function isCatalogMember(m: any, partial = false): m is CatalogMember {
+  return is.string(m?.type) && (is.string(m?.name) || partial);
 }
 
-export const catalogMemberProps = ["description", "info"];
+export const catalogMemberProps: CopyProps[] = [
+  "id",
+  "description",
+  "info",
+  "infoSectionOrder",
+  "shortReport",
+  {
+    v7: "shortReportSections",
+    v8: "shortReportSections",
+    translationFn: (srs: any[]) =>
+      srs.length > 0
+        ? srs.map((shortReport) => {
+            return {
+              name: shortReport.name,
+              content: shortReport.content,
+              show: shortReport.isOpen,
+            };
+          })
+        : undefined,
+  },
+  { v7: "isShown", v8: "show" },
+  "splitDirection",
+  "url",
+  {
+    v7: "opacity",
+    v8: "opacity",
+    translationFn: (opacity: any) =>
+      is.string(opacity)
+        ? parseFloat(opacity)
+        : is.number(opacity)
+        ? opacity
+        : undefined,
+  },
+  "chartDisclaimer",
+  {
+    v7: "rectangle",
+    v8: "rectangle",
+    translationFn: (rectangle: any[]) => {
+      return {
+        west: rectangle[0],
+        south: rectangle[1],
+        east: rectangle[2],
+        north: rectangle[3],
+      };
+    },
+  },
+  "currentTime",
+  {
+    v7: "dateFormat",
+    v8: "dateFormat",
+    translationFn: (dateFormat: any) =>
+      dateFormat.timelineTic ?? dateFormat.currentTime,
+  },
+  "disablePreview",
+  "hideSource",
+  // Note: if v7 initialTimeSource is not "present", "start", or "end" -> set to v8 currentTime property
+  {
+    v7: "initialTimeSource",
+    v8: "initialTimeSource",
+    translationFn: (initialTimeSource: any) =>
+      (({
+        present: "now",
+        start: "start",
+        end: "stop",
+      } as any)[initialTimeSource]),
+  },
+  {
+    v7: "initialTimeSource",
+    v8: "currentTime",
+    translationFn: (initialTimeSource: any) =>
+      !["now", "start", "end"].includes(initialTimeSource)
+        ? initialTimeSource
+        : undefined,
+  },
+  "dataCustodian",
+  {
+    v7: "isLegendVisible",
+    v8: "hideLegendInWorkbench",
+    translationFn: (isLegendVisible: boolean) => !isLegendVisible,
+  },
+];
 
-export function getUnknownProps(o: PlainObject, knownProperties: string[]) {
-  return Object.keys(o).filter((prop) => knownProperties.indexOf(prop) === -1);
+export const imageryLayerProps: CopyProps[] = ["keepOnTop"];
+
+export const catalogMemberPropsIgnore = [
+  "name",
+  "type",
+  "isEnabled",
+  "parents",
+  "legendUrl", // Handled by legend function
+  "legendUrls", // ^^
```

Comment:
`<nit>`This doesn't feel like the right place to skip these. Unless the legend handling is automatic for all model types, skipping shouldn't be automatic.`</nit>`

I'll leave it up to you whether you want to change this or not though.

Reply:
added `legendProps`